### PR TITLE
Feature/create retrieve

### PR DESCRIPTION
**Pull request by Raj-kadevar**

### CODE REVIEW

File: user/models.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from django.db import models
+
+# Create your models here.
```
Comment (Dhruvi-Makwana) [inline user/models.py:3]: remove unnecessary comments.

File: user/urls.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from django.contrib import admin
+from django.urls import path
+
+from user import views
+
+urlpatterns = [
+    path('', views.BookView.as_view(), name="form"),
+    path('index/', views.show,name="index"),
```
Comment (Dhruvi-Makwana) [inline user/urls.py:8]: instead of form write proper name as per functionality.

File: usermanagement/settings.py [hunk 3]
```diff
@@ -37,6 +37,7 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+    'user',
 ]
```
Comment (Dhruvi-Makwana) [inline usermanagement/settings.py:41]: use " " double quote.

File: user/views.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    def get(self, request, *args, **kwargs):
+        form = Reg()
+        return render(request,"user/form.html",{"form":form})
+
+    def post(self, request, *args, **kwargs):
+        entry = Reg(request.POST)
+        if entry.is_valid():
+           user_save = entry.save()
+           # user_save.set_password(user_save.password)
+           user_save.save()
```
Comment (Dhruvi-Makwana) [inline user/views.py:22]: while pushing the code remove comments if not necessary.

File: user/urls.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from django.contrib import admin
+from django.urls import path
+
+from user import views
+
+urlpatterns = [
+    path('', views.BookView.as_view(), name="form"),
```
Comment (Dhruvi-Makwana) [inline user/urls.py:7]: make 'users' as url blank url used to show homepage or index page.

File: user/form.py
Comment (yatishTT) [inline user/form.py:1]: forms.py

File: user/form.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+from django import forms
+from user.models import User
+class Reg(forms.ModelForm):
```
Comment (yatishTT) [inline user/form.py:3]: always add 2 enter between classes

File: user/form.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    password = forms.CharField(widget=forms.PasswordInput())
+    class Meta:
```
Comment (yatishTT) [inline user/form.py:5]: enter

File: user/form.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        model = User
+        fields = ['name', 'email', 'password', 'phone_number', 'address']
+
+    def clean_email(self):
+        email = self.cleaned_data['email']
+        if User.objects.filter(email=email).exists():
+            raise forms.ValidationError("This email is already taken.")
+        return email
+
+    def clean_password(self):
```
Comment (yatishTT) [inline user/form.py:15]: this is wrong remove this field

File: user/form.py (hunk 6 above)
Comment (yatishTT) [inline user/form.py:3]: short forms are not allowed in code

File: user/form.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        password = self.cleaned_data['password']
+        if User.objects.filter(password=password).exists():
+            raise forms.ValidationError("This password is already taken.")
+        return password
+
+    def clean_phone_number(self):
+        number = self.cleaned_data['phone_number']
+        if User.objects.filter(phone_number=number).exists():
+            raise forms.ValidationError("This number is already taken.")
+        elif len(str(number)) != 10:
```
Comment (yatishTT) [inline user/form.py:25]: rhis validation should be on top

File: user/form.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
+from django import forms
+from user.models import User
+class Reg(forms.ModelForm):
+    password = forms.CharField(widget=forms.PasswordInput())
+    class Meta:
+        model = User
+        fields = ['name', 'email', 'password', 'phone_number', 'address']
+
+    def clean_email(self):
```
Comment (yatishTT) [inline user/form.py:9]: where is confirm password

File: user/models.py [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+class User(models.Model):
```
Comment (yatishTT) [inline user/models.py:4]: this should be Django default user model

File: user/templates/user/index.html [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+          height: 100vh;
+          margin: 0;
+          display: grid;
+          grid-template-columns: auto auto;
+          gap: 20px;
+          place-content: center;
+          background: #DAEDFB;
+        }
+        .button {
+          font-family: system-ui, sans-serif;
```
Comment (yatishTT) [inline user/templates/user/index.html:50]: move this css to index.css

File: user/templates/user/index.html [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+          cursor: pointer;
+          border: none;
+          margin: .1em;
+        }
+    </style>
+    <meta charset="UTF-8">
+    <title>Data</title>
+</head>
+<body onload="load()">
+    <script>
```
Comment (yatishTT) [inline user/templates/user/index.html:62]: if not used remove this

File: user/templates/user/index.html [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+    <a style="height:76px" href="{% url 'form' %}"><button class="button" style="--c: #373B44;--b: 5px;--s:12px">Add new</button></a>
+    <table>
+        <tr>
+            <th>Name</th>
+            <th>email</th>
+            <th>phone_number</th>
+            <th>address</th>
+        </tr>
+
+        {% for i in obj %}
```
Comment (yatishTT) [inline user/templates/user/index.html:76]: i = user

Comment (yatishTT) [inline user/templates/user/index.html:76]: i = user

File: user/urls.py (hunk 5 above)
Comment (yatishTT) [reply user/urls.py:7]: UserView name =user, user

File: user/urls.py (hunk 2 above)
Comment (yatishTT) [reply user/urls.py:8]: views.index

File: user/views.py [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+
+from user.form import Reg
+from user.models import User
+
+
+# Create your views here.
+
+class BookView(CreateView):
+
+    def get(self, request, *args, **kwargs):
```
Comment (yatishTT) [inline user/views.py:13]: CreateView does not support get

File: user/views.py [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+        form = Reg()
+        return render(request,"user/form.html",{"form":form})
+
+    def post(self, request, *args, **kwargs):
+        entry = Reg(request.POST)
```
Comment (yatishTT) [inline user/views.py:18]: registration_form = RegistrationForm(request.POST)

File: user/views.py [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+        if entry.is_valid():
+           user_save = entry.save()
+           # user_save.set_password(user_save.password)
```
Comment (yatishTT) [inline user/views.py:21]: password should be set form usermodel on save method if id is None

File: user/views.py [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+           user_save.save()
+        else:
+            form = Reg()
```
Comment (yatishTT) [inline user/views.py:24]: entry.errors

File: user/views.py [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+            error = entry.errors
+            return render(request, "user/form.html", {"err": error,"form":form})
+        return HttpResponseRedirect("index")
+
+def show(request):
```
Comment (yatishTT) [inline user/views.py:29]: users

**Review by yatishTT — CHANGES_REQUESTED** [review]

File: user/form.py [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+
+    def clean_phone_number(self):
+        number = self.cleaned_data['phone_number']
+        if User.objects.filter(phone_number=number).exists():
+            raise forms.ValidationError("This number is already taken.")
+        elif len(str(number)) != 10:
+            raise forms.ValidationError("Please enter 10 digit")
+        else:
+            for i in number:
+                if i not in ('1','2','3','4','5','6','7','8','9','0'):
```
Comment (yatishTT) [inline user/form.py:29]: if number.isdigit()

**Review by yatishTT — CHANGES_REQUESTED** [review]

**Pull request merged**